src/math.zig

```diff
@@ -311,6 +675,104 @@ pub const mat = struct {
         } else @compileError("Expected matrix, found '" ++ @typeName(@TypeOf(a)) ++ "'");
     }
 
+    /// Check if two matrices are exactly equal. For approximate comparison use `equalsApproximately()`
+    pub inline fn equals(a: anytype, b: @TypeOf(a)) bool {
```

Comment:
In general exact equality is pretty non-useful, I would prefer that we make `equal` the name for approximate equality and `equalExact` the name for exact equality.